### PR TITLE
ISLANDORA-1535

### PR DESCRIPTION
**Pull request by ajs6f**
Bringing some visibility to changes in Tuque at the Islandora layer.

### CODE REVIEW

File: includes/admin.form.inc [hunk 1]
```diff
@@ -48,6 +48,21 @@ function islandora_repository_admin(array $form, array &$form_state) {
...
+        'sparql_endpoint_externalized' => array(
+          '#type' => 'checkbox',
+          '#title' => t('Use an external SPARQL Query endpoint'),
+          '#description' => t('If an external query endpoint is used, the repository must be configured to update it.'),
+          '#default_value' => variable_get('sparql_endpoint_externalized', FALSE),
+        ),
+        'sparql_endpoint' => array(
+          '#type' => 'textfield',
+          '#title' => t('SPARQL Query Endpoint'),
+          '#default_value' => variable_get('sparql_endpoint', ""),
```
Comment (DiegoPino) [inline includes/admin.form.inc]: Just minor comment: better if called islandora_sparql_endpoint. Also "islandora_".. for the checkbox variable. We don't know if maybe other contributed module (like a sparql one?) could interfere.

**Comment by ajs6f** [comment]
@DiegoPino Thanks, fixed that!

**Comment by DiegoPino** [comment]
@ajs6f: also islandora_get_tuque_connection() is called all around, which would mean passing the global sparql values also as arguments and modifying a lot of code. Maybe the arguments could be read directly as variables from drupal before passing to tuque instead of passing the as arguments to the constructor? I mean here https://github.com/Islandora/islandora/pull/636/files#diff-6b73cb204f243227c27dae2ca20c0b70R71.

 I know it's not pretty, meaning you would be stuck to one type of tuque connection for the whole site, without been able to override using arguments. Or, just modify 

https://github.com/Islandora/islandora/blob/a23d4934f407560acffb319baf9895a7d37b9a55/islandora.module#L1195

and put your global logic/argument passing there?

Just ideas

**Comment by ajs6f** [comment]
That seems fine to me. I'll make that change.

**Comment by DiegoPino** [comment]
@ajs6f many thanks on the time you are putting on this.

**Comment by ajs6f** [comment]
Hang on, @DiegoPino -- look at https://github.com/Islandora/islandora/pull/636/files#diff-6b73cb204f243227c27dae2ca20c0b70R90. Isn't that going to do the same thing?

**Comment by DiegoPino** [comment]
@ajs6f. Right, you are defaulting to url. But
nope, you are not respecting the global setting. If someone does not check "use" but put's a new url for sparql endpoint, we will end using it anyway. Because
every call to `islandora_get_tuque_connection()` will create an instance of IslandoraTuque with that set to NULL (right now)
So, we need to get that global in islandora_get_tuque_connection() as argument and pass it as argument to the IslandoraTuque constructor. You are reading only the URL from the globals right(and setting the URL if it's null), but not checking if the global is set to true… does that makes sense? Or maybe it's me on a friday :smile:

**Comment by ajs6f** [comment]
@DiegoPino I'm well beyond my limit of understanding for the form builder in use here, but I thought that if "use" is set to FALSE, then the blank for "endpoint" is invisible? And therefore the user cannot put a value into it?

**Comment by DiegoPino** [comment]
@ajs6f: sorry, i'm just too programmed for edge cases = means normal user cases.  Yes, the later case was my thought. But now i'm not sure if unchecking (and hiding as consequence) will remove the url at all on submit or you will  get stuck to an "invisible" global as my fear was...

**Comment by ajs6f** [comment]
Oh, it's cool-- now I think I understand the problem. We have to make sure that if the admin decides to go back to the built-in RI, that wish is respected. I will put in some additional logic to be sure about it and to make it clear.

**Comment by ajs6f** [comment]
@DiegoPino, does that (most recent commit) assuage your concerns? The value for an external endpoint is now only used when the flag for "use an external endpoint" is specifically set. If someone chooses an external endpoint, but then turns the flag back to false, they will be using the internal RI.

**Comment by DiegoPino** [comment]
@ajs6f, yeah, i like it :smile:  it allows, using the current implementation to enforce the globals, but also just override passing directly an url as argument. Al user scenarios covered. Thanks!

**Pull request closed without merging**